var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -128,8 +129,49 @@ def autoreconf(self, spec, prefix):
             autoreconf('-iv')
 
     def configure_args(self):
-        args = ['--with-mpi=%s' % self.spec['mpi'].prefix,
-                'SEQ_CC=%s' % spack_cc]
+        # for convenience
+        spec = self.spec
+
+        # mvapich2 will set the MPI env variables,
+        # but that may not be a given. So set them for safety
+        args = ['--with-mpi=%s' % spec['mpi'].prefix,
+                'SEQ_CC=%s' % spack_cc,
+                'MPICC=%s'  % spec['mpi'].mpicc,
+                'MPICXX=%s' % spec['mpi'].mpicxx,
+                'MPIF77=%s' % spec['mpi'].mpifc,
+                'MPIF90=%s' % spec['mpi'].mpifc,
+                ]
+
+        # setup testing (required at configure time)
+        mpiexec = ''
+        mpiexec_flags = ''
+        # this will be an ordered search.
+        # that isn't great - for example, some HPC machines may use
+        # slurm to manage jobs, but another process launcher to run
+        # e.g., Cray can use aprun + slurm, or mpirun + slurm.
+        mpiexecs = ['srun', 'aprun', 'jsrun', 'mpiexec', 'mpirun']
+        mpiexecs_flags = {'srun': '--cpu-bind=cores -c1 -n',
+                          'aprun': '-n',
+                          'jsrun': '-p',
+                          'mpiexec': '-n',
```

Comment:
That's right - but that ambiguity will never be resolved this way. This only makes it more explicit.  I had a variant to declare the launcher, but removed it because this is maddening - I'll need to apply this stuff in multiple packages,

These are defects in spack more so than the package (as in our discussion about MPI runners)

Reply:
I'm going to try moving mpiexec and mpirun to highest precedence - on this cray, those commands **shouldn't** be in the path... but with the Mvapich2/external module issues, they were ending up in the path. (why I want the MPICC vars set, so it is clearer when the build is going completely bad.)